src/luascript.cpp

```diff
@@ -16064,6 +16069,7 @@ int LuaScriptInterface::luaWeaponRegister(lua_State* L)
 
 		weapon->configureWeapon(it);
 		pushBoolean(L, g_weapons->registerLuaEvent(weapon));
+		*weaponPtr = nullptr; // Remove luascript reference
```

Comment:
Doesn't this leak the valued pointed to?

Reply:
Yeah, I wonder if it is also possible to do the same in this way:
```cpp
weapon = nullptr;
// instead
*weaponPtr = nullptr;
```
because we have already pointed to this pointer in the variable `auto weapon`